src/submit_jobs.py

```diff
@@ -123,17 +132,41 @@ def read_job(job: SparkJob):
         job.repartition()
 
     # Upload extracted files to AWS S3 in Hive-partitioned Parquet
+    aws = AWSClient(logger=session.logger)
+    new_local_files = []
     for job in jobs:
-        job.upload()
+        job_upload_results = job.upload(aws)
+        new_local_files.append(job_upload_results)
+
+    # If any jobs uploaded never-seen-before files, trigger a Glue crawler
+    if any(new_local_files):
+        session.logger.info("New partitions uploaded, triggering Glue crawler")
+        aws.run_and_wait_for_crawler("ccao-data-warehouse-iasworld-crawler")
 
     # Trigger a GitHub workflow to run dbt tests once all jobs are complete
-    github = SharedGitHubSession(gh_pem_path=PATH_GH_PEM)
+    session.logger.info("All file uploads complete, triggering dbt tests")
+    github = GitHubClient(logger=session.logger, gh_pem_path=PATH_GH_PEM)
     github.run_workflow(
         repository="https://api.github.com/repos/ccao-data/data-architecture",
         workflow="test_dbt_models.yaml",
     )
 
-    return session_name
+    # Print table names for extracted tables and total elapsed time. The
+    # get_description() call is to print the description of each table to logs
+    session.logger.info(f"Extracted tables: {', '.join(table_names)}")
+    for job in jobs:
+        job.get_description()
+
+    time_end = time.time()
+    time_duration = str(timedelta(seconds=(time_end - time_start)))
+    session.logger.info(f"Total extraction duration was {time_duration}")
```

Comment:
**[Nitpick, non-blocking]** Can we add a unit to the time duration, for clarity (e.g. `s` or `ms`)?

Reply:
The `timedelta` function actually returns a formatted string like `OH1M14S.100`, so I think we're covered there.